plugins/mtd/fu-self-test.c

```diff
@@ -37,34 +34,26 @@ fu_test_mtd_device_func(void)
 	g_assert_no_error(error);
 	g_assert_true(ret);
 
-	udev_device =
-	    g_udev_client_query_by_sysfs_path(udev_client, "/sys/devices/virtual/mtd/mtd0");
-	if (udev_device == NULL) {
-		g_test_skip("could not find mtdram device");
-		return;
-	}
-	dev_name = g_udev_device_get_property(udev_device, "DEVNAME"); /* nocheck:blocked */
-	if (g_strcmp0(dev_name, "/dev/mtd0") != 0) {
-		g_test_skip("DEVNAME not /dev/mtd0");
-		return;
-	}
-	if (!g_file_test(dev_name, G_FILE_TEST_EXISTS)) {
-		g_test_skip("/dev/mtd0 doesn't exist");
-		return;
-	}
-
 	/* create device */
-	device = g_object_new(FU_TYPE_MTD_DEVICE, "context", ctx, "udev-device", udev_device, NULL);
+	device = g_object_new(FU_TYPE_MTD_DEVICE,
+			      "context",
+			      ctx,
+			      "backend-id",
+			      "/sys/devices/virtual/mtd/mtd0",
+			      NULL);
```

Comment:
don't you need a sysfs tree for the test that replicates this?

Reply:
I need this on top of the prep2 branch really.

Reply:
Which I did, and now it passes :)